src/components/slider/slider.js

```diff
@@ -0,0 +1,210 @@
+   * @extends CreateComponent
+   * @extends InitComponentBySearch
+   * @param {HTMLElement} element The element working as an slider.
+   */
+  constructor(element, options) {
+    super(element, options);
+
+    this.sliderActive = false;
+    this.dragging = false;
+
+    this.track = this.element.querySelector(this.options.selectorTrack);
+    this.filledTrack = this.element.querySelector(this.options.selectorFilledTrack);
+    this.thumb = this.element.querySelector(this.options.selectorThumb);
+    this.input = this.element.querySelector(this.options.selectorInput);
+
+    if (this.element.dataset.sliderInputBox) {
+      this.boundInput = this.element.ownerDocument.querySelector(this.element.dataset.sliderInputBox);
+      this._updateInput();
+      this.boundInput.addEventListener('change', (evt) => { this._handleTextInput(evt); });
+    }
+
+    this._updatePosition();
+
+    this.thumb.addEventListener('mousedown', (evt) => { this._handleMouseDown(evt); });
+    this.element.ownerDocument.addEventListener('mouseup', (evt) => { this._handleMouseUp(evt); });
+    this.element.ownerDocument.addEventListener('mousemove', (evt) => { this._handleMouseMove(evt); });
+    this.thumb.addEventListener('keydown', (evt) => { this._handleKeyDown(evt); });
+    this.track.addEventListener('click', (evt) => { this._handleClick(evt); });
+  }
+
+  _handleMouseDown() {
+    this.sliderActive = true;
+  }
+
+  _handleMouseMove(evt) {
+    if (this.sliderActive === true) {
+      this._updatePosition(evt);
+    }
+  }
+
+  _handleMouseUp() {
+    this.sliderActive = false;
+  }
+
+  _handleKeyDown(evt) {
+    this._updatePosition(evt);
+  }
+
+  _handleClick(evt) {
+    this._updatePosition(evt);
+  }
+
+  _handleTextInput(evt) {
+    this.setValue(evt.target.value);
+  }
+
+  _changeState = (state, detail, callback) => {
+    callback();
+  }
+
+
+  _updatePosition(evt) {
+    const {
+      left,
+      newValue,
+    } = this._calcValue(evt);
+
+
+    if (this.dragging) {
+      return;
+    }
+
+    this.dragging = true;
+
+    requestAnimationFrame(() => {
+      this.dragging = false;
+
+      this.thumb.style.left = `${left}%`;
+      this.filledTrack.style.transform = `scaleX(${left / 100})`;
+      this.input.value = newValue;
+      this._updateInput();
+      this.changeState('slider-value-change', { value: newValue });
+    });
+  }
+
+  _calcValue(evt) {
+    const {
+      value,
+      min,
+      max,
+      step,
+    } = this.getInputProps();
+
+    const range = max - min;
+    const valuePercentage = (((value - min) / range) * 100);
+
+    let left;
+    let newValue;
+    left = valuePercentage;
+    newValue = value;
+
+    if (evt) {
+      const type = evt.type;
+
+      if (type === 'keydown') {
+        let direction;
+        if (evt.which === 40 || evt.which === 37) {
+          direction = 'decreasing';
+        }
+        if (evt.which === 38 || evt.which === 39) {
+          direction = 'increasing';
+        }
```

Comment:
@hellobrian Good call, we can also define `Slider.INCREASING = 1` and `Slider.DECREASING = 1` and use them instead of `increasing`/`decreasing` strings. In this way, we won't need ternary operator to calculate `left`.